<a href="https://colab.research.google.com/github/meherkalsi13-png/ds3001_final_project_group_g/blob/main/DS3001_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Install package and import the dataset to code**

In [6]:
%pip install ucimlrepo

In [12]:
from ucimlrepo import fetch_ucirepo

# fetch dataset
risk_factor_prediction_of_chronic_kidney_disease = fetch_ucirepo(id=857)

# data (as pandas dataframes)
X = risk_factor_prediction_of_chronic_kidney_disease.data.features
y = risk_factor_prediction_of_chronic_kidney_disease.data.targets

# metadata
print(risk_factor_prediction_of_chronic_kidney_disease.metadata, '\n')

# variable information
print(risk_factor_prediction_of_chronic_kidney_disease.variables)

{'uci_id': 857, 'name': 'Risk Factor Prediction of Chronic Kidney Disease', 'repository_url': 'https://archive.ics.uci.edu/dataset/857/risk+factor+prediction+of+chronic+kidney+disease', 'data_url': 'https://archive.ics.uci.edu/static/public/857/data.csv', 'abstract': 'Chronic kidney disease (CKD) is an increasing medical issue that declines the productivity of renal capacities and subsequently damages the kidneys.', 'area': 'Health and Medicine', 'tasks': ['Classification', 'Regression'], 'characteristics': ['Multivariate'], 'num_instances': 200, 'num_features': 28, 'feature_types': ['Real'], 'demographics': [], 'target_col': ['class'], 'index_col': None, 'has_missing_values': 'no', 'missing_values_symbol': None, 'year_of_dataset_creation': 2020, 'last_updated': 'Fri Mar 08 2024', 'dataset_doi': '10.24432/C5WP64', 'creators': ['Md. Ashiqul Islam', 'Shamima Akter'], 'intro_paper': {'ID': 268, 'type': 'NATIVE', 'title': 'Risk Factor Prediction of Chronic Kidney Disease based on Machine L

**Metadata**: '**uci_id**': 857, '**name**': 'Risk Factor Prediction of Chronic Kidney Disease', '**repository_url**': 'https://archive.ics.uci.edu/dataset/857/risk+factor+prediction+of+chronic+kidney+disease', 'data_url': 'https://archive.ics.uci.edu/static/public/857/data.csv',

'**abstract**': 'Chronic kidney disease (CKD) is an increasing medical issue that declines the productivity of renal capacities and subsequently damages the kidneys.',

  '**area**': 'Health and Medicine', '**tasks**': ['Classification', 'Regression'],  '**characteristics**': ['Multivariate'],
  '**num_instances**': 200,
  '**num_features**': 28,
  '**feature_types**': ['Real'],
  '**demographics**': [],
  '**target_col**': ['class'],
  '**index_col**': None,
  '**has_missing_values**': 'no',
  '**missing_values_symbol**': None,
  '**year_of_dataset_creation**': 2020,
  '**last_updated**': 'Fri Mar 08 2024',
  '**dataset_doi**': '10.24432/C5WP64',
  '**creators**': ['Md. Ashiqul Islam', 'Shamima Akter'],


'**intro_paper**':
  '**ID**': 268,
  '**type**': 'NATIVE',
  '**title**': 'Risk Factor Prediction of Chronic Kidney Disease based on Machine Learning Algorithms',
  '**authors**': 'M. Islam, S. Akter, M. Hossen, Sadia Ahmed Keya, Sadia Afrin Tisha, Shahed Hossain',
  '**venue**': 'International Conferences on Information Science and System',
  '**year**': 2020,
  '**journal**': None,
  '**DOI**': None,
  '**URL**': 'https://www.semanticscholar.org/paper/a4b90dd7b9dfaffa9eff1efc6555bbce62a963d7',
  '**sha**': None,
  '**corpus**': None,
  '**arxiv**': None,
  '**mag**': None,
  '**acl**': None,
  '**pmid**': None,
  '**pmcid**': None},

'**additional_info**':
  '**summary**': 'This dataset is real Bangladeshi patient data. The dataset is collected from Enam Medical College, Savar, Dhaka, Bangladesh.\n',
  '**purpose**': None,
  '**funded_by**': None,
  '**instances_represent**': None,
  '**recommended_data_splits**': None,
  '**sensitive_data**': None,
  '**preprocessing_description**': 'This dataset is not pre-processed, if you want to apply a Machine learning Algorithm at first you have to need to pre-process the data',
  '**variable_info**': '1. bp(Diastolic)\n2. bp limit\n3. sg\n4. al\n5. class\n6. rbc\n7. su\n8. pc\n9. pcc\n10. ba\n11.bgr\n12. bu\n13. sod\n14. sc\n15. pot\n16. hemo\n17. pcv\n18. rbcc\n19. wbcc\n20. htn\n21. dm\n22. cad\n23. appet\n24. pe\n25. ane\n26. grf\n27. stage\n28. affected\n29. age',
  '**citation**': "M. A. Islam, S. Akter, M. S. Hossen, S. A. Keya, S. A. Tisha and S. Hossain, 'Risk Factor Prediction of Chronic Kidney Disease based on Machine Learning Algorithms,' 2020 3rd International Conference on Intelligent Sustainable Systems (ICISS), Thoothukudi, India, 2020, pp. 952-957, doi: 10.1109/ICISS49785.2020.9315878."


In [11]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

display(X.head())
display(y.head())



bp (Diastolic)  bp limit             sg     al  rbc   su  pc  pcc  ba  \
0               0         0  1.019 - 1.021  1-Jan    0  < 0   0    0   0   
1               0         0  1.009 - 1.011    < 0    0  < 0   0    0   0   
2               0         0  1.009 - 1.011    ≥ 4    1  < 0   1    0   1   
3               1         1  1.009 - 1.011  3-Mar    0  < 0   0    0   0   
4               0         0  1.015 - 1.017    < 0    0  < 0   0    0   0   

         bgr  ... htn dm cad appet pe ane                grf stage  affected  \
0      < 112  ...   0  0   0     0  0   0          ≥ 227.944    s1         1   
1  112 - 154  ...   0  0   0     0  0   0          ≥ 227.944    s1         1   
2      < 112  ...   0  0   0     1  0   0  127.281 - 152.446    s1         1   
3  112 - 154  ...   0  0   0     0  0   0  127.281 - 152.446    s1         1   
4  154 - 196  ...   0  1   0     1  1   0  127.281 - 152.446    s1         1   

      age  
0    < 12  
1    < 12  
2    < 12  
3    < 12  
4  20-Dec  

[5 rows x 28 columns]

class
0   ckd
1   ckd
2   ckd
3   ckd
4   ckd

No charts were generated by quickchart
